astroquery/splatalogue/build_species_table.py

```diff
@@ -9,21 +9,50 @@
 import os
 import requests
 
+from astropy.config import paths
 
-def data_path(filename):
-    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+
+def data_path(filename: str):
+    """
+    Build the path to save a file, using the ``astropy`` cache
+    directory.
+
+    Parameters
+    ----------
+    filename : str
+        [description]
+
+    Returns
+    -------
+    str
+        Full path to the cache directory
+    """
+    data_dir = os.path.join(paths.get_cache_dir(), 'astroquery/Splatalogue')
     return os.path.join(data_dir, filename)
 
 
-def get_json_species_ids(outfile='species.json'):
+def get_json_species_ids(outfile='splat-species.json'):
     """
-    Load the NRAO Splatalogue form and parse the inputs into a JSON object
+    Uses BeautifulSoup to scrape the NRAO Splatalogue species
+    selector form, and caches the result as JSON. The file
+    is saved to the ``astropy`` cache.
+
+    Parameters
+    ----------
+    outfile : str, optional
+        Name of the output JSON, by default 'splat-species.json'
+
+    Returns
+    -------
+    str
+        Formatted string representation of the JSON object
     """
     import bs4
 
     result = requests.get('https://www.cv.nrao.edu/php/splat/b.php')
     page = bs4.BeautifulSoup(result.content, 'html5lib')
-    sid = page.findAll('select', attrs={'id': 'sid'})[0]
+    # The ID needs to be checked periodically if Splatalogue is updated
+    sid = page.findAll('select', attrs={'id': 'speciesselectbox'})[0]
```

Comment:
so this change is from `sid` to `speciesselectbox`.... can you explain what the difference is?

Reply:
The Splatalogue page changed the id for the table we're looking for from `sid` to `speciesselectbox`—when I was testing I found that `page.findAll` came up with an empty list because the old selector no longer can be found on the page. This uses the correct selector now.